toolkit/tools/specreader/specreader.go

```diff
@@ -200,8 +244,44 @@ func createChroot(workerTar, buildDir, specsDir, srpmsDir string) (chroot *safec
 	return
 }
 
+func findSpecFiles(specsDir string, specListSet map[string]bool) (specFiles []string, err error) {
+	// Find the filepath for each spec in the SPECS directory.
+	if len(specListSet) == 0 {
+		var specSearch string
+		specSearch, err = filepath.Abs(filepath.Join(specsDir, "**/*.spec"))
+		if err != nil {
+			err = fmt.Errorf("invalid spec path: '%s'. Error: %w", specsDir, err)
+			return nil, err
+		}
+		specFiles, err = filepath.Glob(specSearch)
+		if err != nil {
+			err = fmt.Errorf("failed to find *.spec files. Check that %s is the correct directory. Error: %w", specsDir, err)
+			return nil, err
+		}
+		return
+	} else {
+		for specName := range specListSet {
+			var matchingSpecFiles []string
+
+			specSearch := filepath.Join(specsDir, fmt.Sprintf("**/%s.spec", specName))
+			matchingSpecFiles, err = filepath.Glob(specSearch)
```

Comment:
Suggestion: since we're already explicitly mentioning `err` in the returns, we can drop the variable declarations.
```suggestion
		specSearch, err := filepath.Abs(filepath.Join(specsDir, "**/*.spec"))
		if err != nil {
			err = fmt.Errorf("invalid spec path: '%s'. Error: %w", specsDir, err)
			return nil, err
		}
		specFiles, err = filepath.Glob(specSearch)
		if err != nil {
			err = fmt.Errorf("failed to find *.spec files. Check that %s is the correct directory. Error: %w", specsDir, err)
			return nil, err
		}
		return
	} else {
		for specName := range specListSet {
			specSearch := filepath.Join(specsDir, fmt.Sprintf("**/%s.spec", specName))
			matchingSpecFiles, err := filepath.Glob(specSearch)
```